src/node_nanomsg.cc

```diff
@@ -125,6 +130,41 @@ NAN_METHOD(Getsockopt) {
     NanReturnValue(obj);
 }
 
+NAN_METHOD(Setopt) {
+  NanScope();
+
+  int level = args[1].integer;
+  int option = args[2].integer;
+
+  if(option == NN_SOCKET_NAME){
+    utf8 str(args[3]);
+    ret(NanNew<Number>(nn_setsockopt(S, level, option, *str, str.length())));
+  } else {
+    int optval = args[3].integer;
+    ret(NanNew<Number>(nn_setsockopt(S, level, option, &optval, sizeof(optval))));
+  }
+}
+
+NAN_METHOD(Getopt) {
+  NanScope();
+
+  int optval[64];
+  int option = args[2].integer;
+  size_t optsize = sizeof(optval);
+
+  //check if the function succeeds
+  if(nn_getsockopt(S, args[1].integer, option, optval, &optsize) == 0){
```

Comment:
This does work, it's just using `63 * sizeof(int)` bytes more of the stack than it needs to.  It works because in C arrays decay into pointers when passed to functions.  So you're allocating space for 64 ints on the stack, then passing the address of the first element, which gets filled.  Since it's an array of ints, only the first element is ever populated.

If you think about the C ABI, it's not possible to represent 64 ints in a single register on today's processors.  That's why passing an array decays into a pointer.  A kind of syntactic sugar.  `nn_getsockopt` takes a `void*`, which works since `optval` decays into an `int*`.

A few lines down, there is no need to access `optval[0]` since we don't need `optval` to be an array of ints, just an int.

also, does `ret` actually return control flow from the function?